lib/pbench/agent/results.py

```diff
@@ -178,24 +194,33 @@ def copy_result_tb(self):
                 file_count,
                 self.tarball,
             )
-            sys.exit(1)
+            return 1
 
         with open(self.tarball_md5, "r") as md5fp:
             md5sum = md5fp.read().split()[0]
         filename = secure_filename(str(self.tarball))
-        headers = {"filename": filename, "Content-MD5": md5sum}
+        headers = {
+            "filename": filename,
+            "Content-MD5": md5sum,
+            "Authorization": f"Bearer {token}",
+        }
         with self.tarball.open("rb") as f:
             try:
                 response = requests.put(
                     self.upload_url, data=self.read_in_chunks(f), headers=headers
                 )
                 self.logger.info("File uploaded successfully")
             except Exception:
-                self.logger.exception("There was something wrong with your request")
-                sys.exit(1)
-        if not response.status_code == 200:
+                self.logger.exception(
+                    "There was something wrong with file upload request: file: '{}', URL: '{}'",
+                    self.tarball,
+                    self.upload_url,
+                )
+                return 1
```

Comment:
I'll put in a plug for converting this function's error reporting to use exceptions, as well.  In the case at this line, you can use the `raise XXX from exc` form to preserve and chain the original error (in the others, you can raise a new exception); the caller then becomes a `try` instead of an `if`, with the `except` noting the failure and issuing the `continue`.

Reply:
Done